doc/cephfs: update cephfs disaster recovery documentation

vshankar opened the pull request:
This is not complete yet - I will be adding and pushing in more change over the day today.


<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test classic perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-classic/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test crimson perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-crimson/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test signed` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pr-commits/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-commits/config/definitions/ceph-pr-commits.yml)
- `jenkins test make check` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests/config/definitions/ceph-pull-requests.yml)
- `jenkins test make check arm64` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests-arm64/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests-arm64/config/definitions/ceph-pull-requests-arm64.yml)
- `jenkins test submodules` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-pr-submodules/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-submodules/config/definitions/ceph-pr-commits.yml)
- `jenkins test dashboard` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml)
- `jenkins test dashboard cephadm` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-cephadm-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-cephadm-e2e/config/definitions/ceph-dashboard-cephadm-e2e.yml)
- `jenkins test api` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-api/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-api/config/definitions/ceph-pr-api.yml)
- `jenkins test docs` [ReadTheDocs](https://readthedocs.org/projects/ceph/) | [Github Workflow Definition](https://github.com/ceph/ceph/blob/main/.readthedocs.yml)
- `jenkins test ceph-volume all` [Jenkins Jobs](https://jenkins.ceph.com/view/ceph-volume%20PR/) | [Jenkins Jobs Definition](https://github.com/ceph/ceph-build/blob/main/ceph-volume-cephadm-prs/config/definitions/ceph-volume-pr.yml)
- `jenkins test windows` [Jenkins Job](https://jenkins.ceph.com/job/ceph-windows-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-windows-pull-requests/config/definitions/ceph-windows-pull-requests.yml)
- `jenkins test rook e2e` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-orchestrator-rook-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-rook-e2e/config/definitions/ceph-orchestrator-rook-e2e.yml)
</details>

bluikko commented on doc/cephfs/disaster-recovery-experts.rst:
```diff
@@ -73,32 +87,54 @@ file system has or had multiple active MDS daemons.
 
-Use the following command to reset the SessionMap (this will erase all
-sessions):
+   # Sessiona table
```
Seems to have a typo "a", perhaps:

```suggestion
   # Session table
```

bluikko commented on doc/cephfs/disaster-recovery-experts.rst:
```diff
@@ -17,6 +17,11 @@ Advanced: Metadata repair tools
+      tools would complain if they are invoked when the file system is online.
+      In case any of the recovery steps do not run till completion successfully,
+      DO NOT proceeed to run any more recovery steps. At this point, its wise to
+      seek help from experts via mailing-list and/or IRC/slack channels.
```
Slack could start with capital case?

```suggestion
      seek help from experts via mailing-list and/or IRC/Slack channels.
```

vshankar replied:
Of course.

vshankar commented:
Fixed and updated.

Changelog:
- addressed review comments
- added a "Known Limitation and Pitfalls section"
- Warn users about not using alternate pool recovery mechanisms
- Troubleshooting section improvement for MDS stuck in `up:replay`

bluikko commented on doc/cephfs/disaster-recovery-experts.rst:
```diff
@@ -73,32 +87,54 @@ file system has or had multiple active MDS daemons.
-If you know that you need also to reset the other tables, then replace
-``session`` with ``snap`` or ``inode``.
+Note: If the session table have been reset, it would be necessary to remount/restart
+all cephfs clients.
```
Stylize as CephFS?

```suggestion
all CephFS clients.
```

bluikko commented on doc/cephfs/disaster-recovery-experts.rst:
```diff
@@ -207,14 +256,38 @@ operation to delete ancillary data generated during recovery. Use a command of t
+
+#. The data scan commands provide no feedback in terms to estimated time to completion.
+   This is being worked on. See: https://tracker.ceph.com/issues/63191 for details.
+#. It is important to perform a file system scrub after recovery before cephfs clients
```
ditto

```suggestion
#. It is important to perform a file system scrub after recovery before CephFS clients
```

bluikko commented on doc/cephfs/disaster-recovery-experts.rst:
```diff
@@ -207,14 +256,38 @@ operation to delete ancillary data generated during recovery. Use a command of t
+#. In general, its not recommended to change any MDS related settings (max_mds, etc..)
+   while things are broken.
+#. Disaster recovery is a manual process right now. There is plan to automate the
+   recvoery via Disaster Recoery Super Tool. See: https://tracker.ceph.com/issues/71804
```
A few typos here

```suggestion
   recovery via Disaster Recovery Super Tool. See: https://tracker.ceph.com/issues/71804
```

bluikko commented on doc/cephfs/disaster-recovery-experts.rst:
```diff
@@ -207,14 +256,38 @@ operation to delete ancillary data generated during recovery. Use a command of t
+   start using the file system.
+#. In general, its not recommended to change any MDS related settings (max_mds, etc..)
+   while things are broken.
+#. Disaster recovery is a manual process right now. There is plan to automate the
```
Maybe article for "a plan"?

```suggestion
#. Disaster recovery is a manual process right now. There is a plan to automate the
```

vshankar commented:
@ceph/cephfs -- Sounds like there is no major objection and/or additions that is required in this doc update. Anyone?

joscollin commented on doc/cephfs/disaster-recovery-experts.rst:
```diff
@@ -28,6 +34,14 @@ running the following command:
+
+.. prompt:: bash #
+
+   cephfs-journal-tool journal import backup.bin
```
missing mandatory "--rank" argument

joscollin commented on doc/cephfs/disaster-recovery-experts.rst:
```diff
@@ -17,6 +17,12 @@ Advanced: Metadata repair tools
+   system is online. If any of the recovery steps do not complete successfully,
+   DO NOT proceeed to run any more recovery steps. If any recovery step fails,
+   seek help from experts via mailing lists and IRC channels and Slack
+   channels.
```
s/and/or

zdover23 commented:
https://github.com/ceph/ceph/pull/65056 - Tentacle backport
https://github.com/ceph/ceph/pull/65057- Squid backport
https://github.com/ceph/ceph/pull/65058 - Reef backport
